### ISSUE TITLE
Removing 3DES from TLS

### ISSUE THREAD

**Issue by davidben** [mozilla/standards-positions#516]
## Request for Mozilla Position on an Emerging Web Specification

* Specification Title: Remove 3DES from TLS
* Specification or proposal URL: Well, no one really specifies _exact_ TLS cipher suite profiles for browsers anyway, but we marked 3DES as [Recommended=N](https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-4) quite some time ago. :-)
* Caniuse.com URL (optional): 
* Bugzilla URL (optional): https://bugzilla.mozilla.org/show_bug.cgi?id=1227524
* Mozillians who can provide input (optional): @martinthomson @ekr 

### Other information

You all attempted this [previously](https://bugzilla.mozilla.org/show_bug.cgi?id=1386754), though I gather it didn't stick. At the time, 3DES was the price we paid to get rid of RC4. With TLS 1.0 and 1.1 gone, metrics on our end show a whole **0.00%** of TLS connections using 3DES. Probably a good time to try again.
https://groups.google.com/a/chromium.org/g/blink-dev/c/RShdgyaDoX4/m/JikQYHPuBQAJ

**Comment by martinthomson** [comment]
Worth prototyping.

That said, our plan was always to let this go once our TLS 1.0 usage was eliminated.  We're not there on TLS 1.0 yet, so this hadn't come up.

A quick check of Beta telemetry show that we are at 0.01% 3DES.  That is mostly likely tied to TLS 1.0 usage, which is still a noticeable rate.  Nightly is zero, not just 0.00%, but zero connections total across all browser instances that report telemetry.

Our numbers for TLS 1.0 are still such that 0.5% of users encounter a TLS 1.0 server on a daily basis.  We can't draw a direct line to 3DES from that as AES-CBC suites are likely to be well supported. However, as 3DES was the MUST-implement suite for TLS 1.0 and we know from previous investigations that the correlation between TLS 1.0 and 3DES is strong, removing 3DES is not completely without risk.

(I don't think that this needs to go to the dashboard, so I'm going to close this one.)

**Comment by davidben** [comment]
One thing to be aware of with metrics: there are some servers that support AES and 3DES but preferentially pick 3DES. (Whyyyyyy....) Our numbers are after correcting for this, by moving 3DES to a fallback. We did this before we dropped TLS 1.0, and it was quite some time ago, so I don't know how much this inflates 3DES numbers, or whether it's still relevant post-1.0.

**Comment by martinthomson** [comment]
I think that we lost that capability a while back, but we did have a two-pass connection attempt for a while.  And yes, that's why I'm not drawing a direct line between TLS 1.0 and 3DES usage.

As for preferences, that's just inexplicable.  I mean, 3DES is a perfectly good design as far as small-block ciphers from the 70s go, but it's not like AES is *new*.

**Comment by davidben** [comment]
I know, right? :-) I can only assume there's some common mishap with OpenSSL cipher strings or something. Maybe they meant to put 3DES last but put it first instead. Or maybe some explicit SSL3-era cipher config got interpreted as "3DES, RC4, then all the rest" rather than "3DES, RC4, then stop".

I did some digging and looks like we estimated it at 40% [last year](https://groups.google.com/a/chromium.org/g/blink-dev/c/yaJcs4p9LNI/m/haZWzX-UBwAJ), which is surprisingly high.
